### PR TITLE
Support Payment Provider for ne_NP

### PR DESCRIPTION
**Pull request by remotemerge**
### What is the reason for this PR?

This PR supports the `Payment` provider for the `ne_NP` locale.

- [x] Get financial institutions name
- [x] Get Bank Swift Code
- [x] Get bank account number

### Author's checklist

- [x] Follow the [Contribution Guide](https://github.com/FakerPHP/Faker/blob/main/.github/CONTRIBUTING.md)

### Summary of changes

The `Payment.php` and `PaymentTest.php` are new files for the `ne_NP` locale.

### Review checklist

- [x] All checks have passed
- [x] Changes are approved by the maintainer

### CODE REVIEW

File: src/Faker/Provider/ne_NP/Payment.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+namespace Faker\Provider\ne_NP;
+
+class Payment extends \Faker\Provider\Payment
+{
+    /**
+     * List of commercial banks sorted in alphabetical order.
+     *
+     * @var string[]
+     */
+    protected static $commercialBanks = ['Agricultural Development Bank', 'Bank Of Kathmandu', 'Century Commercial Bank', 'Citizens Bank International', 'Civil Bank', 'Everest Bank', 'Global IME Bank', 'Himalayan Bank', 'Kumari Bank', 'Laxmi Bank', 'Machhapuchchhre Bank', 'Mega Bank Nepal', 'Nabil Bank', 'Nepal Bangladesh Bank', 'Nepal Bank', 'Nepal Credit & Commerce Bank', 'Nepal Investment Bank', 'Nepal SBI Bank', 'NIC ASIA Bank', 'NMB Bank', 'Prabhu Bank', 'Prime Commercial Bank', 'Rastriya Banijya Bank', 'Sanima Bank', 'Siddhartha Bank', 'Standard Chartered Bank Nepal', 'Sunrise Bank'];
```
Comment (pimjansen) [inline src/Faker/Provider/ne_NP/Payment.php:12]: Can we put this multiline just like other providers?

File: src/Faker/Provider/ne_NP/Payment.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    /**
+     * List of development banks sorted in alphabetical order.
+     *
+     * @var string[]
+     */
+    protected static $developmentBanks = ['Corporate Development Bank', 'Excel Development Bank', 'Garima Bikas Bank', 'Green Development Bank', 'Jyoti Bikas Bank', 'Kamana Sewa Bikash Bank', 'Karnali Development Bank', 'Lumbini Bikas Bank', 'Mahalaxmi Bikas Bank', 'Miteri Development Bank', 'Muktinath  Bikas Bank', 'Narayani Development Bank', 'Nepal Infrastructure Bank', 'Sahara Bikas Bank', 'Salapa Bikas Bank', 'Saptakoshi Development Bank', 'Shangrila Development Bank', 'Shine Resunga Development Bank', 'Sindhu Bikas Bank'];
```
Comment (pimjansen) [inline src/Faker/Provider/ne_NP/Payment.php:19]: Can we put this multiline just like other providers?

File: src/Faker/Provider/ne_NP/Payment.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+    /**
+     * List of finance companies sorted in alphabetical order.
+     *
+     * @var string[]
+     */
+    protected static $financeCompanies = ['Best Finance Company', 'Capital Merchant Banking & Finance', 'Central Finance', 'Goodwill Finance Company', 'Guheshwori Merchant Banking & Finance', 'Gurkhas Finance', 'ICFC Finance', 'Janaki Finance Company', 'Manjushree Finance', 'Multipurpose Finance Company', 'Nepal Finance', 'Nepal Share Markets', 'Pokhara Finance', 'Progressive Finance', 'Reliance Finance', 'Samriddhi Finance Company', 'Shree Investment Finance Company'];
```
Comment (pimjansen) [inline src/Faker/Provider/ne_NP/Payment.php:26]: Can we put this multiline just like other providers?

File: src/Faker/Provider/ne_NP/Payment.php [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    /**
+     * List of microfinance companies sorted in alphabetical order.
+     *
+     * @var string[]
+     */
+    protected static $microFinances = ['Aatmanirbhar', 'Adarsha', 'Adhikhola', 'Arambha Chautari', 'Asha', 'Aviyan', 'BPW', 'Buddha Jyoti', 'Chhimek', 'Civil', 'CYC Nepal', 'Deprosc', 'Deurali', 'Dhaulagiri', 'First Microfinance', 'Forward Microfinance', 'Ganapati', 'Ghodighoda', 'Global IME', 'Grameen Bikas', 'Gurans', 'Infinity', 'Jalpa Samudayik', 'Janautthan Samudayik', 'Jeevan Bikas', 'Kalika', 'Khaptad', 'Kisan', 'Laxmi', 'Mahila', 'Mahuli', 'Manakamana Smart', 'Manushi', 'Meromicrofinance', 'Mirmire', 'Mithila', 'NADEP', 'National Microfinance', 'Naya Sarathi', 'Nepal Sewa', 'Nerude', 'NESDO Samriddha', 'NIC Asia', 'Nirdhan Utthan', 'NMB', 'Rastra Utthan', 'RMDC', 'RSDC', 'Sabaiko', 'Sadhana', 'Samaj', 'Samata Gharelu', 'Samudayik', 'Sana Kisan Bikas', 'Shrijanshil', 'Summit', 'Super', 'Support', 'Suryodaya', 'Swabalamban', 'Swabhiman', 'Swastik', 'Sworojagar', 'Unique Nepal', 'Unnati Sahakarya', 'Upakar', 'Vijaya', 'WEAN', 'Womi'];
```
Comment (pimjansen) [inline src/Faker/Provider/ne_NP/Payment.php:33]: Can we put this multiline just like other providers?

File: src/Faker/Provider/ne_NP/Payment.php [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+    /**
+     * List of digital wallets sorted in alphabetical order.
+     *
+     * @var string[]
+     */
+    protected static $digitalWallets = ['CellPay', 'CG Pay', 'Chito Paisa', 'DigiPay', 'dPaisa', 'EnetPay', 'eSewa', 'Fonepay', 'GME Pay', 'iCash', 'IME Pay', 'Ipay', 'Khalti', 'Kurakani Pay', 'Lenden', 'Mobalet', 'MOCO', 'Mohar', 'Moru', 'N-Cash', 'Namaste Pay', 'PayTime', 'PayWell', 'PrabhuPAY', 'QPay', 'SajiloPay', 'WePay'];
```
Comment (pimjansen) [inline src/Faker/Provider/ne_NP/Payment.php:40]: Can we put this multiline just like other providers?

File: src/Faker/Provider/ne_NP/Payment.php [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+    /**
+     * List of Swift Codes in alphabetical order.
+     *
+     * @see https://www.theswiftcodes.com/nepal/
+     *
+     * @var string[]
+     */
+    protected static $swiftCodes = ['ADBLNPKA', 'BOKLNPKA', 'CCBNNPKA', 'CIVLNPKA', 'CTZNNPKA', 'EVBLNPKA', 'GLBBNPKA', 'HIMANPKA', 'KMBLNPKA', 'LXBLNPKA', 'MBLNNPKA', 'MBNLNPKA', 'NARBNPKA', 'NBOCNPKA', 'NBOCNPKANRD', 'NEBLNPKA', 'NIBLNPKT', 'NICENPKA', 'NMBBNPKA', 'NPBBNPKA', 'NRBLNPKA', 'NRBLNPKAFED', 'NSBINPKA', 'NSBINPKA001', 'PCBLNPKA', 'PRVUNPKA', 'RBBANPKA', 'SCBLNPKA', 'SIDDNPKA', 'SNMANPKA', 'SRBLNPKA'];
```
Comment (pimjansen) [inline src/Faker/Provider/ne_NP/Payment.php:49]: Can we put this multiline just like other providers?

File: src/Faker/Provider/ne_NP/Payment.php [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+        return static::randomElement(static::$swiftCodes);
+    }
+
+    /**
+     * @example '00454689832792' or 'S49646367883667'
+     */
+    public function bankAccountNumber(): string
+    {
+        $prefixType = self::randomElement(['alpha', 'zero']);
+        $format = $prefixType === 'alpha' ? '[A-Z][1-9]{8,19}' : '[0]{2}[1-9]{7,18}';
```
Comment (pimjansen) [inline src/Faker/Provider/ne_NP/Payment.php:107]: I think its pretty rough to do a string compare for a 50% compare

**Review by pimjansen — CHANGES_REQUESTED** [review]
Overall nice addition but needs some small work. Can you also ensure that for all data additions the source is added?

File: src/Faker/Provider/ne_NP/Payment.php (hunk 1 above)
Comment (bram-pkg) [reply src/Faker/Provider/ne_NP/Payment.php:12]: And please provice a source for the data! :)

File: src/Faker/Provider/ne_NP/Payment.php (hunk 2 above)
Comment (bram-pkg) [reply src/Faker/Provider/ne_NP/Payment.php:19]: Same as my first comment :')

File: src/Faker/Provider/ne_NP/Payment.php (hunk 3 above)
Comment (bram-pkg) [reply src/Faker/Provider/ne_NP/Payment.php:26]: Same as my first comment :')

File: src/Faker/Provider/ne_NP/Payment.php (hunk 4 above)
Comment (bram-pkg) [reply src/Faker/Provider/ne_NP/Payment.php:33]: Same as my first comment :')

File: test/Faker/Provider/ne_NP/PaymentTest.php [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+use Faker\Provider\ne_NP\Payment;
+use Faker\Test\TestCase;
+
+final class PaymentTest extends TestCase
+{
+    public function testCommercialBank(): void
+    {
+        $str = $this->faker->commercialBank();
+        self::assertIsString($str);
```
Comment (bram-pkg) [inline test/Faker/Provider/ne_NP/PaymentTest.php:13]: Please add the specific bank to the test that should be returned. We provide a seed to the provider.

**Review by bram-pkg — CHANGES_REQUESTED** [review]

File: src/Faker/Provider/ne_NP/Payment.php (hunk 7 above)
Comment (remotemerge) [reply src/Faker/Provider/ne_NP/Payment.php]: I changed to boolean.

File: test/Faker/Provider/ne_NP/PaymentTest.php [hunk 9]
```diff
@@ -11,37 +11,68 @@ public function testCommercialBank(): void
...
+        $strNext = $this->faker->commercialBank();
+        self::assertSame($str, $strNext);
     }
 
     public function testDevelopmentBank(): void
     {
         $str = $this->faker->developmentBank();
         self::assertIsString($str);
+
+        $this->faker->seed(1);
```
Comment (bram-pkg) [inline test/Faker/Provider/ne_NP/PaymentTest.php:26]: You do not have to seed Faker, it is done in the `TestCase` class.

**Review by bram-pkg — CHANGES_REQUESTED** [review]

File: test/Faker/Provider/ne_NP/PaymentTest.php (hunk 9 above)
Comment (remotemerge) [reply test/Faker/Provider/ne_NP/PaymentTest.php:26]: @bram-pkg I tested that without seeding. It's not working for sure.

**Comment by remotemerge** [comment]
@bram-pkg @pimjansen it's time to review :)

File: test/Faker/Provider/ne_NP/PaymentTest.php [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    public function testDevelopmentBank(): void
+    {
+        $str = $this->faker->developmentBank();
+        self::assertIsString($str);
+
+        $this->faker->seed(1);
+
+        $strNext = $this->faker->developmentBank();
+        self::assertSame($str, $strNext);
+        echo " $str -> $strNext";
```
Comment (bram-pkg) [inline test/Faker/Provider/ne_NP/PaymentTest.php:30]: Please don't echo in tests.

**Review by bram-pkg — CHANGES_REQUESTED** [review]

File: test/Faker/Provider/ne_NP/PaymentTest.php (hunk 10 above)
Comment (remotemerge) [reply test/Faker/Provider/ne_NP/PaymentTest.php]: @bram-pkg removed.

**Review by bram-pkg — APPROVED** [review]
Looks good to me now!

**Review by pimjansen — APPROVED** [review]

**Comment by pimjansen** [comment]
Thanks!

**Pull request merged**